File: selenium-webdriver-cucumber/pages/flight_page.rb
require File.join(File.dirname(__FILE__), "abstract_page.rb")

class FlightPage < AbstractPage

  def initialize(driver)
    super(driver, "") # <= TEXT UNIQUE TO THIS PAGE
  end

  def select_trip_type(trip_type)
    driver.find_elements(:name => "tripType").each { |elem| elem.click && break if elem.attribute("value") == trip_type && elem.attribute("type") == "radio" }
  end

  def select_depart_from(from_port)
    Selenium::WebDriver::Support::Select.new(driver.find_element(:name, "fromPort")).select_by(:text, from_port)
  end

  def select_arrive_at(to_port)
    Selenium::WebDriver::Support::Select.new(driver.find_element(:name, "toPort")).select_by(:text, to_port)
  end

  def select_depart_day(depart_day)
    Selenium::WebDriver::Support::Select.new(driver.find_element(:id, "departDay")).select_by(:text, depart_day)
  end

  def select_depart_month(depart_month)
    Selenium::WebDriver::Support::Select.new(driver.find_element(:id, "departMonth")).select_by(:text, depart_month)
  end

  def select_return_day(return_day)
    Selenium::WebDriver::Support::Select.new(driver.find_element(:id, "returnDay")).select_by(:text, return_day)
  end

  def select_return_month(return_month)
    Selenium::WebDriver::Support::Select.new(driver.find_element(:id, "returnMonth")).select_by(:text, return_month)
  end

  def click_continue
    driver.find_element(:xpath,"//input[@value='Continue']").click
  end

end

File: selenium-webdriver-python-unittest/pages/flight_page.py
from pages.abstract_page import AbstractPage

from selenium import webdriver
from selenium.webdriver.support.ui import Select

class FlightPage(AbstractPage):

  def select_trip_type(self, trip_type):
    self.driver.find_element_by_xpath("//input[@name='tripType' and @value='" + trip_type + "']").click()

  def select_depart_from(self, from_port):
    Select(self.driver.find_element_by_name("fromPort")).select_by_visible_text(from_port)

  def select_arrive_at(self, to_port):
    Select(self.driver.find_element_by_name("toPort")).select_by_visible_text(to_port)

  def select_depart_day(self, depart_day):
    Select(self.driver.find_element_by_name("departDay")).select_by_visible_text(depart_day)

  def select_depart_month(self, depart_month):
    Select(self.driver.find_element_by_name("departMonth")).select_by_visible_text(depart_month)

  def select_return_day(self, return_day):
    Select(self.driver.find_element_by_name("returnDay")).select_by_visible_text(return_day)

  def select_return_month(self, return_month):
    Select(self.driver.find_element_by_id("returnMonth")).select_by_visible_text(return_month)

  def click_continue(self):
    self.driver.find_element_by_xpath("//input[@value='Continue']").click()
